site/app/views/forum/ForumThreadView.php

```diff
@@ -590,162 +590,150 @@ public function showAlteredDisplayList($threads, $filtering, $thread_id, $catego
-					$return = "";
-					$used_active = false; //used for the first one if there is not thread_id set
-					$current_user = $this->core->getUser()->getId();
-					$display_thread_ids = $this->core->getUser()->getGroup() <= 2;
-
-					$activeThreadAnnouncement = false;
-					$activeThreadTitle = "";
-					$function_date = 'date_format';
-					$activeThread = array();
-
-					foreach($threads as $thread){
-						$first_post = $this->core->getQueries()->getFirstPostForThread($thread["id"]);
-						if(is_null($first_post)) {
-							// Thread without any posts(eg. Merged Thread)
-							$first_post = array('content' => "");
-							$date = null;
-						} else {
-							$date = DateUtils::parseDateTime($first_post['timestamp'], $this->core->getConfig()->getTimezone());
-						}
-						if($thread['merged_thread_id'] != -1){
-							// For the merged threads
-							$thread['status'] = 0;
-						}
+    public function displayThreadList($threads, $filtering, &$activeThreadAnnouncement, &$activeThreadTitle, &$activeThread, $thread_id_p, $current_categories_ids)
+    {
+        $return = "";
+        $used_active = false; //used for the first one if there is not thread_id set
+        $current_user = $this->core->getUser()->getId();
+        $display_thread_ids = $this->core->getUser()->getGroup() <= 2;
 
-						$class = "thread_box";
-						// $current_categories_ids should be subset of $thread["categories_ids"]
-						$issubset = (count(array_intersect($current_categories_ids, $thread["categories_ids"])) == count($current_categories_ids));
-						if(((isset($_REQUEST["thread_id"]) && $_REQUEST["thread_id"] == $thread["id"]) || $thread_id_p == $thread["id"] || $thread_id_p == -1) && !$used_active && $issubset) {
-							$class .= " active";
-							$used_active = true;
-							$activeThreadTitle = ($display_thread_ids ? "({$thread['id']}) " : '') . $thread["title"];
-							$activeThread = $thread;
-							if($thread["pinned"])
-								$activeThreadAnnouncement = true;
-							if($thread_id_p == -1)
-								$thread_id_p = $thread["id"];
-						}
-						if(!$this->core->getQueries()->viewedThread($current_user, $thread["id"])){
-							$class .= " new_thread";
-						}
-						if($thread["deleted"]) {
-							$class .= " deleted";
-						}
-						//fix legacy code
-						$titleDisplay = html_entity_decode($thread['title'], ENT_QUOTES | ENT_HTML5, 'UTF-8');
+        $activeThreadAnnouncement = false;
+        $activeThreadTitle = "";
+        $function_date = 'date_format';
+        $activeThread = array();
 
-						//replace tags from displaying in sidebar
-						$first_post_content = str_replace("[/code]", "", str_replace("[code]", "", strip_tags($first_post["content"])));
-						$temp_first_post_content = preg_replace('#\[url=(.*?)\](.*?)(\[/url\])#', '$2', $first_post_content);
+        $thread_content = Array();
```

Comment:
Let's convert use of `array( key => value )` to `[ key => value ]`

Reply:
Okay!